functests/validation/validation.go

```diff
@@ -0,0 +1,232 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/utils/pointer"
+
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	perfv1alpha1 "github.com/openshift-kni/performance-addon-operators/pkg/apis/performance/v1alpha1"
+	clientmachineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	sriovv1 "github.com/openshift/sriov-network-operator/pkg/apis/sriovnetwork/v1"
+
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+)
+
+var (
+	machineConfigSelector     string
+	sctpMachineConfigSelector string
+	sriovNodeSelector         string
+	performanceNodeSelector   string
+)
+
+func init() {
+	machineConfigSelector = os.Getenv("MACHINE_CONFIG_SELECTOR")
+	if machineConfigSelector == "" {
+		machineConfigSelector = "machineconfiguration.openshift.io/role=worker-cnf"
+	}
+
+	sctpMachineConfigSelector = machineConfigSelector
+
+	sriovNodeSelector = os.Getenv("SRIOV_NODE_SELECTOR")
+	if sriovNodeSelector == "" {
+		sriovNodeSelector = "node-role.kubernetes.io/worker-cnf="
+	}
+
+	performanceNodeSelector = os.Getenv("PERFORMANCE_NODE_SELECTOR")
+	if performanceNodeSelector == "" {
+		performanceNodeSelector = "node-role.kubernetes.io/worker-cnf="
+	}
+}
+
+var _ = Describe("validation", func() {
+	Context("General", func() {
+		It("should report all machine config pools are in ready status", func() {
+			mcp := &clientmachineconfigv1.MachineConfigPoolList{}
+			err := client.Client.List(context.TODO(), mcp, &goclient.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, mcItem := range mcp.Items {
+				Expect(mcItem.Status.MachineCount).To(Equal(mcItem.Status.ReadyMachineCount))
+			}
+		})
+
+		It("should have one machine config pool with the requested label", func() {
+			mcp := &clientmachineconfigv1.MachineConfigPoolList{}
+			selector, err := labels.Parse(machineConfigSelector)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = client.Client.List(context.TODO(), mcp, &goclient.ListOptions{LabelSelector: selector})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(mcp.Items)).To(Equal(1))
+		})
+
+		It("should have a openshift-ovn-kubernetes namespace", func() {
```

Comment:
Can you put this under "[ovn]" context? This will allow us to skip / filter.
Also, I would replace the It definition with It("Should have openshift-ovn as the default snd") or something like that, to make clear why this is failing.